src/devices/sound/sn76477.cpp

```diff
@@ -1996,7 +1998,7 @@ void sn76477_device::sound_stream_update(sound_stream &stream, stream_sample_t *
 		 */
 		*buffer++ = (((voltage_out - OUT_LOW_CLIP_THRESHOLD) / (OUT_CENTER_LEVEL_VOLTAGE - OUT_LOW_CLIP_THRESHOLD)) - 1) * 32767;
 
-		if (LOG_WAV && LOG_WAV_ENABLED_ONLY && !m_enable)
+		if (LOG_WAV && (!LOG_WAV_ENABLED_ONLY || (LOG_WAV_ENABLED_ONLY && !m_enable)))
```

Comment:
You could write the expression as `(LOG_WAV && (!LOG_WAV_ENABLED_ONLY || !m_enable))` and make it more obvious what it does, but why did you change the condition?  Is it out of sync with something else?

Reply:
Indeed your minified expression is the same of that I've proposed but I find more human readable and obvious the long form.
I changed the condition because it was wrong. If LOG_WAV_ENABLED is defined as false the device will never write anything in the log wave file.

Reply:
Ah right, the condition makes sense.  It's generally better to avoid excessive levels of nesting and unnecessarily long lines.  now that I know what it's supposed to do I'd write it as `(LOG_WAV && (!m_enable || !LOG_WAV_ENABLED_ONLY))` – then it reads more naturally, like “log WAV and either enable is low or log when enabled only is unset“.

Reply:
Changed. Please look at my latest commit, thank you.